ccl/benchccl/rttanalysisccl: TestBenchmarkExpectation failed

ccl/benchccl/rttanalysisccl.TestBenchmarkExpectation [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3520559&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3520559&tab=artifacts#/) on master @ [fb5b6bbf05909b141c1a0860f6e344e44c04082a](https://github.com/cockroachdb/cockroach/commits/fb5b6bbf05909b141c1a0860f6e344e44c04082a):


```
=== RUN   TestBenchmarkExpectation
=== CONT  TestBenchmarkExpectation
    validate_benchmark_data.go:73: see the --rewrite flag to re-run the benchmarks and adjust the expectations
--- FAIL: TestBenchmarkExpectation (123.75s)
=== RUN   TestBenchmarkExpectation/BenchmarkAlterPrimaryRegion
    validate_benchmark_data.go:137: /go/src/github.com/cockroachdb/cockroach/artifacts/go-build2256276149/b1284/rttanalysisccl.test --test.run=^$ --test.bench=BenchmarkAlterPrimaryRegion --test.benchtime=1x --test.v
=== CONT  TestBenchmarkExpectation/BenchmarkAlterPrimaryRegion
    validate_benchmark_data.go:150: 
        	Error Trace:	validate_benchmark_data.go:150
        	            				validate_benchmark_data.go:88
        	Error:      	Received unexpected error:
        	            	exit status 1
        	Test:       	TestBenchmarkExpectation/BenchmarkAlterPrimaryRegion
        	Messages:   	failed to wait for command /go/src/github.com/cockroachdb/cockroach/artifacts/go-build2256276149/b1284/rttanalysisccl.test --test.run=^$ --test.bench=BenchmarkAlterPrimaryRegion --test.benchtime=1x --test.v
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterTableLocality_alter_from_regional_by_table_to_rbr071731049
    test_log_scope.go:80: use -show-logs to present logs inline
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterRegions_alter_populated_database_add_region972422151
    test_log_scope.go:80: use -show-logs to present logs inline
    rtt_analysis_bench.go:131: -- test log scope end --
    rtt_analysis_bench.go:54: -- test log scope end --
BenchmarkAlterTableLocality/alter_from_regional_by_table_to_rbr-16            	       1	3484869111 ns/op	        22.00 roundtrips
BenchmarkAlterTableLocality/alter_from_rbr_to_regional_by_table
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterTableLocality_alter_from_rbr_to_regional_by_table090447284
    test_log_scope.go:80: use -show-logs to present logs inline
    rtt_analysis_bench.go:131: -- test log scope end --
    rtt_analysis_bench.go:54: -- test log scope end --
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterRegions_alter_populated_database_add_region353324986
    test_log_scope.go:80: use -show-logs to present logs inline
    rtt_analysis_bench.go:131: -- test log scope end --
    rtt_analysis_bench.go:54: -- test log scope end --
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterTableLocality_alter_from_rbr_to_regional_by_table488879463
    test_log_scope.go:80: use -show-logs to present logs inline
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterSurvivalGoals_alter_populated_database_from_region_to_zone324426154
    test_log_scope.go:80: use -show-logs to present logs inline
    rtt_analysis_bench.go:131: -- test log scope end --
    rtt_analysis_bench.go:54: -- test log scope end --
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterRegions_alter_populated_database_add_region488249237
    test_log_scope.go:80: use -show-logs to present logs inline
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterTableLocality_alter_from_rbr_to_regional_by_table036884890
    test_log_scope.go:80: use -show-logs to present logs inline
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterSurvivalGoals_alter_populated_database_from_region_to_zone000154053
    test_log_scope.go:80: use -show-logs to present logs inline
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterTableLocality_alter_from_rbr_to_regional_by_table264397301
    test_log_scope.go:80: use -show-logs to present logs inline
    rtt_analysis_bench.go:131: -- test log scope end --
    rtt_analysis_bench.go:54: -- test log scope end --
BenchmarkAlterRegions/alter_populated_database_add_region-16     	       1	 732970963 ns/op	        20.00 roundtrips
    --- FAIL: TestBenchmarkExpectation/BenchmarkAlterPrimaryRegion (71.73s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestBenchmarkExpectation PKG=./pkg/ccl/benchccl/rttanalysisccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-experience postamar
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/benchccl/rttanalysisccl.TestBenchmarkExpectation [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3535881&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3535881&tab=artifacts#/) on master @ [d92fe4e01688a072e9415c9b910bac3562615841](https://github.com/cockroachdb/cockroach/commits/d92fe4e01688a072e9415c9b910bac3562615841):


```
=== RUN   TestBenchmarkExpectation
=== CONT  TestBenchmarkExpectation
    validate_benchmark_data.go:73: see the --rewrite flag to re-run the benchmarks and adjust the expectations
--- FAIL: TestBenchmarkExpectation (123.61s)
=== RUN   TestBenchmarkExpectation/BenchmarkAlterPrimaryRegion
=== CONT  TestBenchmarkExpectation/BenchmarkAlterPrimaryRegion
    validate_benchmark_data.go:137: /go/src/github.com/cockroachdb/cockroach/artifacts/go-build3521023932/b1282/rttanalysisccl.test --test.run=^$ --test.bench=BenchmarkAlterPrimaryRegion --test.benchtime=1x --test.v
=== CONT  TestBenchmarkExpectation/BenchmarkAlterPrimaryRegion
    validate_benchmark_data.go:150: 
        	Error Trace:	validate_benchmark_data.go:150
        	            				validate_benchmark_data.go:88
        	Error:      	Received unexpected error:
        	            	exit status 1
        	Test:       	TestBenchmarkExpectation/BenchmarkAlterPrimaryRegion
        	Messages:   	failed to wait for command /go/src/github.com/cockroachdb/cockroach/artifacts/go-build3521023932/b1282/rttanalysisccl.test --test.run=^$ --test.bench=BenchmarkAlterPrimaryRegion --test.benchtime=1x --test.v
    rtt_analysis_bench.go:131: -- test log scope end --
    rtt_analysis_bench.go:54: -- test log scope end --
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterSurvivalGoals_alter_populated_database_from_region_to_zone988085967
    test_log_scope.go:80: use -show-logs to present logs inline
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterRegions_alter_populated_database_add_region527484536
    test_log_scope.go:80: use -show-logs to present logs inline
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterTableLocality_alter_from_rbr_to_regional_by_table523654974
    test_log_scope.go:80: use -show-logs to present logs inline
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterSurvivalGoals_alter_populated_database_from_region_to_zone004259894
    test_log_scope.go:80: use -show-logs to present logs inline
    rtt_analysis_bench.go:131: -- test log scope end --
    rtt_analysis_bench.go:54: -- test log scope end --
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterRegions_alter_populated_database_add_region471303543
    test_log_scope.go:80: use -show-logs to present logs inline
    rtt_analysis_bench.go:131: -- test log scope end --
    rtt_analysis_bench.go:54: -- test log scope end --
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logBenchmarkAlterTableLocality_alter_from_rbr_to_regional_by_table175116677
    test_log_scope.go:80: use -show-logs to present logs inline
    rtt_analysis_bench.go:131: -- test log scope end --
    rtt_analysis_bench.go:54: -- test log scope end --
BenchmarkAlterSurvivalGoals/alter_populated_database_from_region_to_zone-16     	       1	 442348187 ns/op	        41.00 roundtrips
    --- FAIL: TestBenchmarkExpectation/BenchmarkAlterPrimaryRegion (73.02s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestBenchmarkExpectation PKG=./pkg/ccl/benchccl/rttanalysisccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-experience postamar
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/bench/rttanalysis/validate_benchmark_data.go
+++ b/pkg/bench/rttanalysis/validate_benchmark_data.go
@@ -22,7 +22,6 @@ import (
 	"sort"
 	"strconv"
 	"strings"
-	"sync"
 	"testing"
 
 	"github.com/cockroachdb/cockroach/pkg/testutils"
@@ -74,44 +73,32 @@ func RunBenchmarkExpectationTests(t *testing.T) {
 		}
 	}()
 
-	var g sync.WaitGroup
-	run := func(b string) {
-		tf := func(t *testing.T) {
-			flags := []string{
-				"--test.run=^$",
-				"--test.bench=" + b,
-				"--test.benchtime=1x",
-			}
-			if testing.Verbose() {
-				flags = append(flags, "--test.v")
-			}
-			results := runBenchmarks(t, flags...)
+	for _, b := range benchmarks {
+		flags := []string{
+			"--test.run=^$",
+			"--test.bench=" + b,
+			"--test.benchtime=1x",
+		}
+		if testing.Verbose() {
+			flags = append(flags, "--test.v")
+		}
+		results := runBenchmarks(t, flags...)
 
-			for _, r := range results {
-				exp, ok := expectations.find(r.name)
-				if !ok {
-					t.Logf("no expectation for benchmark %s, got %d", r.name, r.result)
-					continue
-				}
-				if !exp.matches(r.result) {
-					t.Errorf("fail: expected %s to perform KV lookups in [%d, %d], got %d",
-						r.name, exp.min, exp.max, r.result)
-				} else {
-					t.Logf("success: expected %s to perform KV lookups in [%d, %d], got %d",
-						r.name, exp.min, exp.max, r.result)
-				}
+		for _, r := range results {
+			exp, ok := expectations.find(r.name)
+			if !ok {
+				t.Logf("no expectation for benchmark %s, got %d", r.name, r.result)
+				continue
+			}
+			if !exp.matches(r.result) {
+				t.Errorf("fail: expected %s to perform KV lookups in [%d, %d], got %d",
+					r.name, exp.min, exp.max, r.result)
+			} else {
+				t.Logf("success: expected %s to perform KV lookups in [%d, %d], got %d",
+					r.name, exp.min, exp.max, r.result)
 			}
 		}
-		g.Add(1)
-		go func() {
-			defer g.Done()
-			t.Run(b, tf)
-		}()
 	}
-	for _, b := range benchmarks {
-		run(b)
-	}
-	g.Wait()
 }
 
 func getBenchmarks(t *testing.T) (benchmarks []string) {

